tidb-cloud/v8.5-performance-highlights.md

```diff
@@ -0,0 +1,191 @@
+
+- Memory limits: in releases earlier than TiDB 8.1, all transaction mutations are held in memory throughout the transaction lifecycle, which strains resources and reduces performance. For operations involving millions of rows, this could lead to excessive memory usage and, in some cases, Out of Memory (OOM) errors when resources are insufficient.
+
+- Performance slowdowns: managing large in-memory buffers relies on red-black trees, which introduces computational overhead. As buffers grew, their operations slowed down due to the *O(N log N)* complexity inherent in these data structures.
+
+### Solution
+
+These challenges highlighted a clear opportunity to improve scalability, reduce complexity, and enhance reliability. With the rise of modern data workloads, TiDB introduces the [Pipelined DML](https://docs.pingcap.com/tidb/stable/system-variables#tidb_dml_type-new-in-v800) feature, designed to change how large transactions are handled and improve resource utilization and overall performance.
+
+### Test environment
+
+- Cluster topology: TiDB (16 vCPU, 32 GiB) \* 1 + TiKV (16 vCPU, 32 GiB) \* 3
+- Dataset: YCSB non-clustered table with 10 million rows (about 10 GiB of data). Primary keys are selectively removed in certain tests to isolate and evaluate the impact of hotspot patterns.
+- Workload: DML operations including `INSERT`, `UPDATE`, and `DELETE`.
+
+### Test results
+
+The execution speed increases by 2x, the maximum TiDB memory usage decreases by 50%, and the TiKV write flow becomes much smoother.
+
+- Latency (in seconds)
+
+    | Workload (10 GiB) | Standard DML | Pipelined DML | Improvement |
+    | --- |  --- |  --- |  --- |
+    | YCSB-insert-10M | 368 | 159 | 131.45% |
+    | YCSB-update-10M | 255 | 131 | 94.66% |
+    | YCSB-delete-10M | 136 | 42 | 223.81% |
+
+- TiDB memory usage peak (GiB)
+
+    | Workload (10 GiB) | Standard DML | Pipelined DML | Reduction |
+    | --- |  --- |  --- |  --- |
+    | YCSB-insert-10M | 25.8 | 12 | 53.49% |
+    | YCSB-update-10M | 23.1 | 12.9 | 44.16% |
+    | YCSB-delete-10M | 10.1 | 8.08 | 20.00% |
+
+## TiKV scaling performance
+
+### Challenge
+
+Horizontal scaling is a core capability of TiKV, enabling the system to scale in or out as needed. As business demands grow and the number of tenants increases, TiDB clusters experience rapid growth in databases, tables, and data volume. Scaling out TiKV nodes quickly becomes essential to maintaining service quality.
+
+In some scenarios, TiDB hosts a large number of databases and tables. When these tables are small or empty, TiKV accumulates a significant number of tiny regions, especially when the number of tables grows to a large scale (such as 1 million or more). These small regions introduce a substantial maintenance burden, increase resource overhead, and reduce efficiency.
```

Comment:
> TiKV accumulates a significant number of tiny regions

Does "region" here refer to ["Region"](https://docs.pingcap.com/tidbcloud/tidb-cloud-glossary#region)? If so, please capitalize it as "Region".



> TiDB Region
>
> The basic unit of data in TiDB. TiKV divides the Key-Value space into a series of consecutive Key segments, and each segment is called a Region.